### PR TITLE
lapack/gonum: add Dgesv

### PR DESCRIPTION
**Pull request by vladimir-ch**
More for completeness than utility because Dgesv doesn't have a trans parameter to solve `A^T*X = B`.

Please take a look.

### CODE REVIEW

File: lapack/testlapack/dgesv.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"golang.org/x/exp/rand"
```
Comment (kortschak) [inline lapack/testlapack/dgesv.go:12]: Blank between.

**Review by kortschak — COMMENTED** [review]
A nit.

File: lapack/testlapack/dgesv.go (hunk 1 above)
Comment (vladimir-ch) [reply lapack/testlapack/dgesv.go:12]: Thanks, done. I've also added a check of `ok` in the test.

**Review by kortschak — APPROVED** [review]

**Comment by vladimir-ch** [comment]
I took a look at Dgesvx but that's a much more complex call, testing it would be a hell and there are unimplemented dependencies, so I won't go there.

**Pull request merged**